library/debug.c

```diff
@@ -91,6 +91,10 @@ void mbedtls_debug_print_msg( const mbedtls_ssl_context *ssl, int level,
         str[ret]     = '\n';
         str[ret + 1] = '\0';
     }
+    else
+    {
+        str[DEBUG_BUF_SIZE - 2] = '\n';
```

Comment:
`vsnprintf()` man page says "If an output error is encountered, a negative value is returned", so this should only be done if `ret >= DEBUG_BUF_SIZE`; if `ret < 0` we should do `str[0] = '\n'` and `str[1] = '\0'`

Since we're just truncating it might be easier to say something like `if (ret >= DEBUG_BUF_SIZE) ret = DEBUG_BUF_SIZE - 2` and have the `str[ret] = '\n'` (etc) code handle it

Reply:
I just pushed out the new code. Can you please review it again?